inst/include/sys/Makevars

```diff
@@ -1,6 +1,6 @@
 STANHEADERS_SRC = $(shell "$(R_HOME)/bin$(R_ARCH_BIN)/Rscript" -e "message()" -e "cat(system.file('include', 'src', package = 'StanHeaders', mustWork = TRUE))" -e "message()" | grep "StanHeaders")
 
-PKG_CPPFLAGS = -I"../inst/include" -I"$(STANHEADERS_SRC)" -DBOOST_DISABLE_ASSERTS -DEIGEN_NO_DEBUG -DBOOST_MATH_OVERFLOW_ERROR_POLICY=errno_on_error
+PKG_CPPFLAGS = -I"../inst/include" -I"$(STANHEADERS_SRC)" -DBOOST_DISABLE_ASSERTS -DEIGEN_NO_DEBUG -DBOOST_MATH_OVERFLOW_ERROR_POLICY=errno_on_error -DUSE_STANC3
```

Comment:
`USE_STANC3` should only be defined if the C++ code is generated by `stanc3`. This is ok when `rstan 2.26` is released, but the transition to CRAN may require releasing https://github.com/stan-dev/rstan/pull/912 first that's compatible with the older version of `rstan`. So, if `USE_STANC3` is defined, Math v4+ headers will be exposed and break the C++ code generated by `stanc2`.

**Solution:** While the development version of `rstan` defines `USE_STANC3` in the generated C++ code, it's safe to define it here only `if (utils::packageVersion("rstan") >= 2.26)`.

Reply:
@andrjohns A variant of the following should work:
```bash
"$(R_HOME)/bin$(R_ARCH_BIN)/Rscript" -e "cat(ifelse(utils::packageVersion('rstan') >= 2.26, '-DUSE_STANC3', ''))"
```